repose-aggregator/core/repose-core-api/src/test/scala/org/openrepose/core/container/config/ContainerSchemaTest.scala

```diff
@@ -166,55 +212,102 @@ class ContainerSchemaTest extends FunSpec with Matchers {
+    it("should accept a config where the Via Configuration is defined at both the deployment and cluster config elements") {
       val config =
         """<repose-container xmlns='http://docs.openrepose.org/repose/container/v2.0'>
           |    <deployment-config>
           |        <deployment-directory>/var/repose</deployment-directory>
           |        <artifact-directory>/usr/share/repose/filters</artifact-directory>
+          |        <via-configuration content="test"/>
           |    </deployment-config>
           |
           |    <cluster-config cluster-id="foo">
-          |        <ssl-configuration>
-          |            <key-password>password</key-password>
-          |        </ssl-configuration>
+          |        <via-configuration content="override"/>
           |    </cluster-config>
           |</repose-container>""".stripMargin
-      val exception = intercept[SAXParseException] {
-        validator.validateConfigString(config)
-      }
-      exception.getLocalizedMessage should include("When defining a new keystore to use, the keystore and key passwords are required")
+      validator.validateConfigString(config)
     }
 
-    it("should successfully validate a config where ssl-configuration is partially patched and is in the base") {
+    it("should accept a config where the Via Configuration disables the via header") {
       val config =
         """<repose-container xmlns='http://docs.openrepose.org/repose/container/v2.0'>
           |    <deployment-config>
           |        <deployment-directory>/var/repose</deployment-directory>
           |        <artifact-directory>/usr/share/repose/filters</artifact-directory>
-          |        <ssl-configuration>
-          |            <keystore-filename>keystore.jks</keystore-filename>
-          |            <keystore-password>password</keystore-password>
-          |            <key-password>password</key-password>
-          |        </ssl-configuration>
           |    </deployment-config>
           |
           |    <cluster-config cluster-id="foo">
-          |        <ssl-configuration>
-          |            <key-password>other-password</key-password>
-          |        </ssl-configuration>
+          |        <via-configuration enabled="false"/>
           |    </cluster-config>
           |</repose-container>""".stripMargin
       validator.validateConfigString(config)
     }
   }
+
+  private val coreSpringProvider = CoreSpringProvider.getInstance()
+  coreSpringProvider.initializeCoreContext("/etc/repose", false)
+  private val reposeVersion = coreSpringProvider.getCoreContext.getEnvironment.getProperty(
+    ReposeSpringProperties.stripSpringValueStupidity(ReposeSpringProperties.CORE.REPOSE_VERSION))
+  describe("deprecated schema validation") {
+    if (reposeVersion.startsWith("8.")) {
+      it("should reject a config where cluster IDs are not unique across all patches") {
```

Comment:
These were the original tests. I was just keeping them in the suite until they were obsoleted by age. These were the only tests that originally used the old `via` attribute.